domain.go

```diff
@@ -1832,10 +1832,11 @@ func (d *Domain) ListAllInterfaceAddresses(src uint) ([]DomainInterface, error)
 
 		for k := 0; k < numAddr; k++ {
 			ifaces[i].Addrs[k] = DomainIPAddress{}
-			ifaces[i].Addrs[k].Type = int(addrSlice[k]._type)
-			ifaces[i].Addrs[k].Addr = C.GoString(addrSlice[k].addr)
-			ifaces[i].Addrs[k].Prefix = uint(addrSlice[k].prefix)
-
+			if addSlice[k] != nil {
```

Comment:
Unfortunately, this doesn't work either. This time however it panics with `runtime error: index out of range`. `numAddr` doesn't seem to represent the actual length of `addrSlice`.